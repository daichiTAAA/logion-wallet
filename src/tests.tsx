import { ReactElement } from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import reactRenderer, { ReactTestRenderer, ReactTestRendererJSON, act as reactAct } from 'react-test-renderer';

export function render(element: ReactElement): null | ReactTestRendererJSON | ReactTestRendererJSON[] {
    const component = reactRenderer.create(element);
    return component.toJSON();
}

export function renderer(element: ReactElement): ReactTestRenderer {
    return reactRenderer.create(element);
}

export function shallowRender<E extends ReactElement>(element: ReactElement): E {
    const renderer = createRenderer();
    renderer.render(element);
    return renderer.getRenderOutput();
}

export function mockAccount(address: string, name: string) {
    return {
        address,
        meta: {
            name
        }
    };
}

export { reactAct as act };
