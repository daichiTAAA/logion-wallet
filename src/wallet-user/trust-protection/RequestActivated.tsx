import React from 'react';

import { ProtectionRequest } from "../../legal-officer/Types";
import LegalOfficerInfo from "../../component/LegalOfficerInfo";

import { legalOfficerByAddress, checkActivation } from "./Model";
import { Button } from "react-bootstrap";

export interface Props {
    request: ProtectionRequest,
}

export default function RequestActivated(props: Props) {

    const [confirmButtonEnabled, setConfirmButtonEnabled] = React.useState(props.request.status === "PENDING");

    return (
        <>
            <h2>My Legal Officers</h2>
            <p>
                Your Logion Trust Protection is active. You are now protected by
            </p>
            <ul>
                {
                    props.request.decisions
                        .map(decision => {
                            const legalOfficer = legalOfficerByAddress(decision.legalOfficerAddress);
                            return (
                                <li key={legalOfficer.address}>
                                    <LegalOfficerInfo legalOfficer={legalOfficer}/>
                                </li>
                            );
                        })
                }
            </ul>
            {
                props.request.isRecovery &&
                <p>
                    You may initiate the actual recovery of account {props.request.addressToRecover}.
                </p>
            }
            {
                // This button is a safety net in case the same call at the previous step failed.
                // In most cases, it will not show
                confirmButtonEnabled &&
                <>
                    <p>Your protection does not seem to be confirmed yet.</p>
                    <Button id="btnConfirmProtection" onClick={() => {
                        checkActivation(props.request)
                            .then(() => setConfirmButtonEnabled(false))
                    }}>
                        Confirm protection
                    </Button>
                </>
            }
        </>
    );
}
