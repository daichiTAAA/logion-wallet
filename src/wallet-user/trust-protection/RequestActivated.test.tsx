import React from 'react';

import { shallowRender } from "../../tests";

import RequestActivated from './RequestActivated';
import {
    ACTIVATED_PROTECTION_REQUEST,
    PENDING_PROTECTION_REQUEST,
    ACTIVATED_RECOVERY_REQUEST, PENDING_RECOVERY_REQUEST
} from './TestData';

test("activated protection request", () => {
    const tree = shallowRender(<RequestActivated request={ ACTIVATED_PROTECTION_REQUEST }/>)
    expect(tree).toMatchSnapshot();
});

test("pending protection request", () => {
    const tree = shallowRender(<RequestActivated request={ PENDING_PROTECTION_REQUEST }/>)
    expect(tree).toMatchSnapshot();
});

test("activated recovery request", () => {
    const tree = shallowRender(<RequestActivated request={ ACTIVATED_RECOVERY_REQUEST }/>)
    expect(tree).toMatchSnapshot();
});

test("pending recovery request", () => {
    const tree = shallowRender(<RequestActivated request={ PENDING_RECOVERY_REQUEST }/>)
    expect(tree).toMatchSnapshot();
});
