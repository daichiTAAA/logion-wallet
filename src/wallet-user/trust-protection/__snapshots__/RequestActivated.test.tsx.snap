// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`activated protection request 1`] = `
<React.Fragment>
  <h2>
    My Legal Officers
  </h2>
  <p>
    Your Logion Trust Protection is active. You are now protected by
  </p>
  <ul>
    <li>
      <LegalOfficerInfo
        legalOfficer={
          Object {
            "address": "5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty",
            "details": "SELARL ADRASTEE
Gare des Brotteaux
14, place Jules Ferry
69006 LYON
France",
            "name": "Guillaume",
          }
        }
      />
    </li>
    <li>
      <LegalOfficerInfo
        legalOfficer={
          Object {
            "address": "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY",
            "details": "Modero Bruxelles
Central administration 
Nijverheidslaan 1
B - 1853 Grimbergen (Strombeek-Bever)
Belgium",
            "name": "Patrick",
          }
        }
      />
    </li>
  </ul>
</React.Fragment>
`;

exports[`activated recovery request 1`] = `
<React.Fragment>
  <h2>
    My Legal Officers
  </h2>
  <p>
    Your Logion Trust Protection is active. You are now protected by
  </p>
  <ul>
    <li>
      <LegalOfficerInfo
        legalOfficer={
          Object {
            "address": "5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty",
            "details": "SELARL ADRASTEE
Gare des Brotteaux
14, place Jules Ferry
69006 LYON
France",
            "name": "Guillaume",
          }
        }
      />
    </li>
    <li>
      <LegalOfficerInfo
        legalOfficer={
          Object {
            "address": "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY",
            "details": "Modero Bruxelles
Central administration 
Nijverheidslaan 1
B - 1853 Grimbergen (Strombeek-Bever)
Belgium",
            "name": "Patrick",
          }
        }
      />
    </li>
  </ul>
  <p>
    You may initiate the actual recovery of account 
    toRecover
    .
  </p>
</React.Fragment>
`;

exports[`pending protection request 1`] = `
<React.Fragment>
  <h2>
    My Legal Officers
  </h2>
  <p>
    Your Logion Trust Protection is active. You are now protected by
  </p>
  <ul>
    <li>
      <LegalOfficerInfo
        legalOfficer={
          Object {
            "address": "5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty",
            "details": "SELARL ADRASTEE
Gare des Brotteaux
14, place Jules Ferry
69006 LYON
France",
            "name": "Guillaume",
          }
        }
      />
    </li>
    <li>
      <LegalOfficerInfo
        legalOfficer={
          Object {
            "address": "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY",
            "details": "Modero Bruxelles
Central administration 
Nijverheidslaan 1
B - 1853 Grimbergen (Strombeek-Bever)
Belgium",
            "name": "Patrick",
          }
        }
      />
    </li>
  </ul>
  <React.Fragment>
    <p>
      Your protection does not seem to be confirmed yet.
    </p>
    <Button
      active={false}
      disabled={false}
      id="btnConfirmProtection"
      onClick={[Function]}
      variant="primary"
    >
      Confirm protection
    </Button>
  </React.Fragment>
</React.Fragment>
`;

exports[`pending recovery request 1`] = `
<React.Fragment>
  <h2>
    My Legal Officers
  </h2>
  <p>
    Your Logion Trust Protection is active. You are now protected by
  </p>
  <ul>
    <li>
      <LegalOfficerInfo
        legalOfficer={
          Object {
            "address": "5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty",
            "details": "SELARL ADRASTEE
Gare des Brotteaux
14, place Jules Ferry
69006 LYON
France",
            "name": "Guillaume",
          }
        }
      />
    </li>
    <li>
      <LegalOfficerInfo
        legalOfficer={
          Object {
            "address": "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY",
            "details": "Modero Bruxelles
Central administration 
Nijverheidslaan 1
B - 1853 Grimbergen (Strombeek-Bever)
Belgium",
            "name": "Patrick",
          }
        }
      />
    </li>
  </ul>
  <p>
    You may initiate the actual recovery of account 
    toRecover
    .
  </p>
  <React.Fragment>
    <p>
      Your protection does not seem to be confirmed yet.
    </p>
    <Button
      active={false}
      disabled={false}
      id="btnConfirmProtection"
      onClick={[Function]}
      variant="primary"
    >
      Confirm protection
    </Button>
  </React.Fragment>
</React.Fragment>
`;
